### ISSUE TITLE
Questions about the big-math library: Capabilities and Future.

### ISSUE THREAD

**Issue by PowerUser1234** [eobermuhlner/big-math#60]
-In terms of base 10 positive intetgers, is there a maximum positive integer that your library can load into memory and operate with?

-In terms of base 10  positive decimals (only), is there a minimum minute value that your library can load into memory and operate with?

-Full completion of the project Valhalla for Java along with OpenJDK promises value types and primitive types.  I have gathered that it is possible to set up mathematics with the big-math library so that every operation does not require the precision to be submitted along as well, but that can be managed from one place within scope before the beginning of mathematics.  If Valhalla allows operator inclusion with its new primitive types and other new advantages, are the big-math project developers likely going to be persuaded to incorporated into big-math? **Operators like +,-,*, both modes of /, %, i++, i--, --i, ++i, +=, -=, *=, /= both modes, %=**

**Comment by eobermuhlner** [comment]
The `big-math` library is built on top of the standard Java `BigDecimal` and has the same memory limitations.

Yes - I would probably provide Valhalla operators as soon as they are available. Depending on the details of Valhalla I would consider providing them in a separate library.

**Comment by PowerUser1234** [comment]
The way that the Java OpenJDK original BigInteger and Bigdecimal method arithmetic methods presently work, every time you want to do an arithmetic operation, you need to include the MathContext, containing precision, each time.  Admittedly, it should be easy to create a Factory class to mask and once submit a MathContext object so that arithmetic methods could be called more easily.

Does your library include the ability to just create and submit the Mathcontent/precision just once, at the top of a codespace, inside a "Factory" class, to generate BigIntegers and BigDecimals from there, so that you don't have to submit two parameters for every arithmetic call every time?

**Comment by eobermuhlner** [comment]
Yes, there is a `DefaultBigDecimalMath` that does not need to pass a `MathContext` with every call.
This class is used in the `kotlin-big-math` library: https://github.com/eobermuhlner/kotlin-big-math for the operators.

**Comment by PowerUser1234** [comment]
Is there one of those for BigInteger as well?

**Comment by eobermuhlner** [comment]
BigInteger does not need this, since they do not use a MathContext to
specify a precision.

**Comment by PowerUser1234** [comment]
Why does the big-math library's BigInteger not need a MathContext? What does it use to specify the maximum number of figures?

**Comment by eobermuhlner** [comment]
The BigInteger operations always calculate the precision from the input
arguments
(roughly max of arguments precisions for add/subtract and the sum of
argument precisions for multiplication).

BigDecimal operations optionally do the same thing.
The add/subtract/mutliply methods exist in a variant without MathContext.
Only for the divide operation the MathContext is very strongly recommended.

**Comment by PowerUser1234** [comment]
If I wish to use the big-math library for OpenJDK Java, I need an **Integer** class, a **Decimal** class, and a **Calculator** class.  What are the names of these 3 (three) classes?

**Comment by eobermuhlner** [comment]
I do not understand this question.

Who needs these 3 classes?

JDK does not have or need them (there is an Integer class - but I think you
mean something else).
big-math does not have or need them.

You can use big-math library with any standard OpenJDK >= 1.8 and you do
not need any additional libraries or any additional classes.

**Comment by PowerUser1234** [comment]
I have not mentioned classes, I have mentioned types of Numbers.  What classes does big-math have that corresponds to an integer number, decimal number, and a calculator class for these two types of numbers?

**Comment by eobermuhlner** [comment]
I still do not understand the question.
big-math does not have its own number types or anything (with the exception
of BigComplex which is a composition of 2 BigDecimal).

Did you read the big-math documentation and try a little hello world
example ?
